crates/transaction-pool/src/validate/eth.rs

```diff
@@ -442,8 +464,31 @@ where
             )
         }
 
-        // Checks for max cost
+        #[cfg(not(feature = "optimism"))]
         let cost = transaction.cost();
+
+        #[cfg(feature = "optimism")]
+        let cost = {
+            let block = match self.client.block_by_number_or_tag(BlockNumberOrTag::Latest) {
+                Ok(block) => block.unwrap_or_default(),
+                Err(err) => {
+                    return TransactionValidationOutcome::Error(*transaction.hash(), Box::new(err))
+                }
+            };
+
+            let cost_addition = match L1BlockInfo::try_from(&block) {
+                Ok(info) => {
+                    info.calculate_tx_l1_cost(transaction.input(), transaction.is_deposit())
```

Comment:
isn't `info.calculate_tx_l1_cost(.., true)` a noop here since we reject any incoming deposit transactions earlier anyway? does it make sense to simplify `calculate_tx_l1_cost` fn to accept only `input`?

Reply:
No I don't believe so - this calculates other transactions l1 cost using the latest blocks l1 block info. Since the deposit tx is pre-paid, it doesn't need this calculation.